Compilation fails on CUDA 10.2

cmake -DCUDA_ARCH_LIST=6.1 -DCUDA_TOOLKIT_ROOT_DIR=/opt/nvidia/hpc_sdk/Linux_x86_64/21.9/cuda/10.2 -DCMAKE_INSTALL_PREFIX=/home/ari/anaconda3/envs/ct2-281 -DWITH_CUDA=ON -DWITH_MKL=ON -DWITH_DNNL=OFF -DWITH_OPENBLAS=OFF -DENABLE_CPU_DISPATCH=ON ..

make VERBOSE=1;

.......

```
[ 14%] Building NVCC (Device) object CMakeFiles/ctranslate2.dir/src/ops/ctranslate2_generated_mean_gpu.cu.o
cd /ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops && /usr/local/bin/cmake -E make_directory /ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops/.
cd /ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops && /usr/local/bin/cmake -D verbose:BOOL=1 -D build_configuration:STRING=Release -D generated_file:STRING=/ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops/./ctranslate2_generated_mean_gpu.cu.o -D generated_cubin_file:STRING=/ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops/./ctranslate2_generated_mean_gpu.cu.o.cubin.txt -P /ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops/ctranslate2_generated_mean_gpu.cu.o.Release.cmake
-- Removing /ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops/./ctranslate2_generated_mean_gpu.cu.o
/usr/local/bin/cmake -E rm -f /ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops/./ctranslate2_generated_mean_gpu.cu.o
-- Generating dependency file: /ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops/ctranslate2_generated_mean_gpu.cu.o.NVCC-depend
/opt/nvidia/hpc_sdk/Linux_x86_64/21.9/cuda/10.2/bin/nvcc -M -D__CUDACC__ /ai/onmt/onmt-2200/CTranslate2-281/src/ops/mean_gpu.cu -o /ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops/ctranslate2_generated_mean_gpu.cu.o.NVCC-depend -ccbin /usr/bin/c++ -m64 -Dctranslate2_EXPORTS -DCT2_X86_BUILD -DCT2_WITH_CPU_DISPATCH -DCT2_WITH_MKL -DMKL_ILP64 -DCT2_WITH_CUDA -DSTACK_LINE_READER_BUFFER_SIZE=1024 -DTHRUST_HOST_SYSTEM=THRUST_HOST_SYSTEM_CPP -DTHRUST_DEVICE_SYSTEM=THRUST_DEVICE_SYSTEM_CUDA -Xcompiler ,\"-Wall\",\"-Wextra\",\"-ffast-math\",\"-m64\",\"-fPIC\",\"-O3\",\"-DNDEBUG\" -std=c++14 -Xcompiler=-fopenmp -gencode arch=compute_61,code=sm_61 -DNVCC -I/opt/nvidia/hpc_sdk/Linux_x86_64/21.9/cuda/10.2/include -I/ai/onmt/onmt-2200/CTranslate2-281/src -I/ai/onmt/onmt-2200/CTranslate2-281/third_party -I/opt/intel/mkl/include -I/ai/onmt/onmt-2200/CTranslate2-281/include -I/ai/onmt/onmt-2200/CTranslate2-281/third_party/spdlog/include -I/ai/onmt/onmt-2200/CTranslate2-281/third_party/cpu_features/include -I/ai/onmt/onmt-2200/CTranslate2-281/third_party/thrust -I/ai/onmt/onmt-2200/CTranslate2-281/third_party/thrust/dependencies/cub
-- Generating temporary cmake readable file: /ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops/ctranslate2_generated_mean_gpu.cu.o.depend.tmp
/usr/local/bin/cmake -D input_file:FILEPATH=/ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops/ctranslate2_generated_mean_gpu.cu.o.NVCC-depend -D output_file:FILEPATH=/ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops/ctranslate2_generated_mean_gpu.cu.o.depend.tmp -D verbose=1 -P /usr/local/share/cmake-3.18/Modules/FindCUDA/make2cmake.cmake
-- Copy if different /ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops/ctranslate2_generated_mean_gpu.cu.o.depend.tmp to /ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops/ctranslate2_generated_mean_gpu.cu.o.depend
/usr/local/bin/cmake -E copy_if_different /ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops/ctranslate2_generated_mean_gpu.cu.o.depend.tmp /ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops/ctranslate2_generated_mean_gpu.cu.o.depend
-- Removing /ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops/ctranslate2_generated_mean_gpu.cu.o.depend.tmp and /ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops/ctranslate2_generated_mean_gpu.cu.o.NVCC-depend
/usr/local/bin/cmake -E rm -f /ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops/ctranslate2_generated_mean_gpu.cu.o.depend.tmp /ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops/ctranslate2_generated_mean_gpu.cu.o.NVCC-depend
-- Generating /ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops/./ctranslate2_generated_mean_gpu.cu.o
/opt/nvidia/hpc_sdk/Linux_x86_64/21.9/cuda/10.2/bin/nvcc /ai/onmt/onmt-2200/CTranslate2-281/src/ops/mean_gpu.cu -c -o /ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops/./ctranslate2_generated_mean_gpu.cu.o -ccbin /usr/bin/c++ -m64 -Dctranslate2_EXPORTS -DCT2_X86_BUILD -DCT2_WITH_CPU_DISPATCH -DCT2_WITH_MKL -DMKL_ILP64 -DCT2_WITH_CUDA -DSTACK_LINE_READER_BUFFER_SIZE=1024 -DTHRUST_HOST_SYSTEM=THRUST_HOST_SYSTEM_CPP -DTHRUST_DEVICE_SYSTEM=THRUST_DEVICE_SYSTEM_CUDA -Xcompiler ,\"-Wall\",\"-Wextra\",\"-ffast-math\",\"-m64\",\"-fPIC\",\"-O3\",\"-DNDEBUG\" -std=c++14 -Xcompiler=-fopenmp -gencode arch=compute_61,code=sm_61 -DNVCC -I/opt/nvidia/hpc_sdk/Linux_x86_64/21.9/cuda/10.2/include -I/ai/onmt/onmt-2200/CTranslate2-281/src -I/ai/onmt/onmt-2200/CTranslate2-281/third_party -I/opt/intel/mkl/include -I/ai/onmt/onmt-2200/CTranslate2-281/include -I/ai/onmt/onmt-2200/CTranslate2-281/third_party/spdlog/include -I/ai/onmt/onmt-2200/CTranslate2-281/third_party/cpu_features/include -I/ai/onmt/onmt-2200/CTranslate2-281/third_party/thrust -I/ai/onmt/onmt-2200/CTranslate2-281/third_party/thrust/dependencies/cub
/ai/onmt/onmt-2200/CTranslate2-281/src/ops/mean_gpu.cu(17): error: explicit type is missing ("int" assumed)
          detected during instantiation of "void ctranslate2::ops::Mean::compute<D,T>(const ctranslate2::StorageView &, ctranslate2::dim_t, ctranslate2::dim_t, ctranslate2::dim_t, ctranslate2::StorageView &) const [with D=ctranslate2::Device::CUDA, T=float]" 
(58): here

/ai/onmt/onmt-2200/CTranslate2-281/src/ops/mean_gpu.cu(17): error: expected a ";"
          detected during instantiation of "void ctranslate2::ops::Mean::compute<D,T>(const ctranslate2::StorageView &, ctranslate2::dim_t, ctranslate2::dim_t, ctranslate2::dim_t, ctranslate2::StorageView &) const [with D=ctranslate2::Device::CUDA, T=float]" 
(58): here

/ai/onmt/onmt-2200/CTranslate2-281/src/ops/mean_gpu.cu(18): error: a class or namespace qualified name is required
          detected during instantiation of "void ctranslate2::ops::Mean::compute<D,T>(const ctranslate2::StorageView &, ctranslate2::dim_t, ctranslate2::dim_t, ctranslate2::dim_t, ctranslate2::StorageView &) const [with D=ctranslate2::Device::CUDA, T=float]" 
(58): here

/ai/onmt/onmt-2200/CTranslate2-281/src/ops/mean_gpu.cu(18): error: global-scope qualifier (leading "::") is not allowed
          detected during instantiation of "void ctranslate2::ops::Mean::compute<D,T>(const ctranslate2::StorageView &, ctranslate2::dim_t, ctranslate2::dim_t, ctranslate2::dim_t, ctranslate2::StorageView &) const [with D=ctranslate2::Device::CUDA, T=float]" 
(58): here

/ai/onmt/onmt-2200/CTranslate2-281/src/ops/mean_gpu.cu(18): error: expected a ";"
          detected during instantiation of "void ctranslate2::ops::Mean::compute<D,T>(const ctranslate2::StorageView &, ctranslate2::dim_t, ctranslate2::dim_t, ctranslate2::dim_t, ctranslate2::StorageView &) const [with D=ctranslate2::Device::CUDA, T=float]" 
(58): here

/ai/onmt/onmt-2200/CTranslate2-281/src/ops/mean_gpu.cu(28): error: identifier "temp_storage" is undefined
          detected during instantiation of "void ctranslate2::ops::Mean::compute<D,T>(const ctranslate2::StorageView &, ctranslate2::dim_t, ctranslate2::dim_t, ctranslate2::dim_t, ctranslate2::StorageView &) const [with D=ctranslate2::Device::CUDA, T=float]" 
(58): here

/ai/onmt/onmt-2200/CTranslate2-281/src/ops/mean_gpu.cu(28): error: identifier "BlockReduce" is undefined
          detected during instantiation of "void ctranslate2::ops::Mean::compute<D,T>(const ctranslate2::StorageView &, ctranslate2::dim_t, ctranslate2::dim_t, ctranslate2::dim_t, ctranslate2::StorageView &) const [with D=ctranslate2::Device::CUDA, T=float]" 
(58): here

7 errors detected in the compilation of "/tmp/tmpxft_00006811_00000000-6_mean_gpu.cpp1.ii".
-- Removing /ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops/./ctranslate2_generated_mean_gpu.cu.o
/usr/local/bin/cmake -E rm -f /ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops/./ctranslate2_generated_mean_gpu.cu.o
CMake Error at ctranslate2_generated_mean_gpu.cu.o.Release.cmake:280 (message):
  Error generating file
  /ai/onmt/onmt-2200/CTranslate2-281/build/CMakeFiles/ctranslate2.dir/src/ops/./ctranslate2_generated_mean_gpu.cu.o


CMakeFiles/ctranslate2.dir/build.make:6070: recipe for target 'CMakeFiles/ctranslate2.dir/src/ops/ctranslate2_generated_mean_gpu.cu.o' failed
make[2]: *** [CMakeFiles/ctranslate2.dir/src/ops/ctranslate2_generated_mean_gpu.cu.o] Error 1
make[2]: Leaving directory '/ai/onmt/onmt-2200/CTranslate2-281/build'
CMakeFiles/Makefile2:177: recipe for target 'CMakeFiles/ctranslate2.dir/all' failed
make[1]: *** [CMakeFiles/ctranslate2.dir/all] Error 2
make[1]: Leaving directory '/ai/onmt/onmt-2200/CTranslate2-281/build'
Makefile:148: recipe for target 'all' failed
make: *** [all] Error 2

```

When configuring & building with CUDA 11.4 everything works fine:
 cmake -DCUDA_ARCH_LIST=6.1 -DCUDA_TOOLKIT_ROOT_DIR=/opt/nvidia/hpc_sdk/Linux_x86_64/21.9/cuda/11.4 -DCMAKE_INSTALL_PREFIX=/home/ari/anaconda3/envs/ct2-281 -DWITH_CUDA=ON -DWITH_MKL=ON -DWITH_DNNL=OFF -DWITH_OPENBLAS=OFF -DENABLE_CPU_DISPATCH=ON ..

I have been working on to port CTranslate2 to ppc64le, but this issue is reproducible on vanilla x86 as seen above. Working CUDA 10.2 support would be appreciated!

--- a/src/ops/mean_gpu.cu
+++ b/src/ops/mean_gpu.cu
@@ -1,5 +1,7 @@
 #include "ctranslate2/ops/mean.h"
 
+#include <cub/cub.cuh>
+
 #include "type_dispatch.h"
 #include "cuda/helpers.h"
 

--- a/src/ops/multinomial_gpu.cu
+++ b/src/ops/multinomial_gpu.cu
@@ -3,6 +3,7 @@
 #include <chrono>
 #include <memory>
 
+#include <cub/cub.cuh>
 #include <curand_kernel.h>
 
 #include "type_dispatch.h"

